conans/client/graph/graph_binaries.py

```diff
@@ -182,6 +182,15 @@ def _evaluate_node(self, node, build_mode, remotes, update):
             node.build_allowed = True
             node.binary = BINARY_BUILD if not node.cant_build else BINARY_INVALID
 
+        if node.binary == BINARY_BUILD:
+            conanfile = node.conanfile
+            if conanfile.vendor and not conanfile.conf.get("tools.graph:vendor", choices=("build",)):
+                node.conanfile.info.invalid = f"The package '{conanfile.ref}' is a vendoring one, " \
+                                              f"needs to be built from source, but it " \
+                                              "didn't enable 'tools.graph:vendor=build' to compute " \
```

Comment:
```suggestion
                                              "didn't enable '{conanfile.ref.name}/*:tools.graph:vendor=build' to compute " \
```

Reply:
Wdyt @memsharded

Reply:
We have talked about it, better keep it for all, as it there is more than one, it will be necessary to define all.